### PR TITLE
Add bytes type

### PR DESCRIPTION
**Pull request by vedantroy**
Implements the bytes datatype for Automerge + 3 tests for it (2 frontend, 1 backend). (Do we need more?)

### CODE REVIEW

File: automerge-protocol/src/serde_impls/op.rs [hunk 1]
```diff
@@ -155,7 +155,7 @@ impl<'de> Deserialize<'de> for Op {
                         Some(ScalarValue::F32(n)) => Ok(OpType::Inc(n as i64)),
                         Some(ScalarValue::Counter(n)) => Ok(OpType::Inc(n)),
                         Some(ScalarValue::Timestamp(n)) => Ok(OpType::Inc(n)),
-                        Some(ScalarValue::Str(s)) => {
+                        Some(ScalarValue::Str(s) | ScalarValue::Bytes(s)) => {
```
Comment (jeffa5) [inline automerge-protocol/src/serde_impls/op.rs:158]: This is giving a build error.

File: automerge-backend/src/columnar.rs [hunk 2]
```diff
@@ -429,9 +429,11 @@ impl<'a> Iterator for ValueIterator<'a> {
             }
             v if v % 16 == VALUE_TYPE_BYTES => {
                 let len = v >> 4;
-                let _data = self.val_raw.read_bytes(len).ok()?;
-                unimplemented!()
-                //Some((amp::Value::Bytes(data))
+                let data = self.val_raw.read_bytes(len).ok()?;
+                Some(amp::ScalarValue::Bytes(
+                    // This unwrap is fine b/c we encode as valid UTF-8 when serializing
+                    String::from_utf8(data.into()).unwrap(),
```
Comment (jeffa5) [inline automerge-backend/src/columnar.rs:435]: This is only safe if we know we encoded it but someone else could have so I think this should return the error.

**Review by jeffa5 — CHANGES_REQUESTED** [review]
I'm not totally aware of the context for adding this as it seems like users could always just base64 encode stuff themselves and use `Str` to store it but I'm still fine with this being added.

File: automerge-protocol/src/serde_impls/op.rs (hunk 1 above)
Comment (vedantroy) [reply automerge-protocol/src/serde_impls/op.rs]: Fixed

File: automerge-backend/src/columnar.rs (hunk 2 above)
Comment (vedantroy) [reply automerge-backend/src/columnar.rs]: Closing b/c new version stores raw bytes (no string).

File: automerge-protocol/src/lib.rs [hunk 3]
```diff
@@ -214,6 +217,13 @@ impl ScalarValue {
                     datatype,
                 }),
             },
+            (DataType::Bytes, ScalarValue::Bytes(bytes)) => Ok(ScalarValue::Bytes(bytes.clone())),
+            (DataType::Bytes, v) => Err(error::InvalidScalarValue {
+                raw_value: self.clone(),
+                expected: "a base64 encoded string of bytes".to_string(),
```
Comment (jeffa5) [inline automerge-protocol/src/lib.rs:223]: I think this needs updating since we aren't using b64 anymore

**Review by jeffa5 — CHANGES_REQUESTED** [review]
Looking good, CI is passing now. I think the previous failure was due to match patterns.

Just one bit of b64 left to remove.

File: automerge-protocol/src/lib.rs (hunk 3 above)
Comment (vedantroy) [reply automerge-protocol/src/lib.rs]: Fixed

**Pull request merged**